do

```diff
@@ -50,12 +52,52 @@ def update_foretold(args):
         log.info(f"Saved Foretold to {dest}")
 
 
+def _web_export(
+    comment: str,
+    export_regions: List[str],
+    region_dataset: RegionDataset,
+    models: Path,
+    rates: Path,
+    hopkins: Path,
+    foretold: Path,
+    output_dir: Path,
+    date_resample: str,
+) -> None:
+    ex = WebExport(date_resample, comment=comment)
+
+    simulation_specs: pd.DataFrame = pd.read_hdf(models, "simulations")
+    models_df: pd.DataFrame = pd.read_hdf(models, "new_fraction")
+    # TODO: unify indexing
+    rates_df: pd.DataFrame = pd.read_csv(rates, index_col="CodeM49")
+    hopkins_df: pd.DataFrame = pd.read_csv(hopkins, index_col="Code", parse_dates=["Date"])
+    foretold_df: pd.DataFrame = pd.read_csv(foretold, index_col="Code", parse_dates=["Date"])
+
+    for code in export_regions:
+        reg = region_dataset[code]
+        ex.new_region(
```

Comment:
I would strongly prefer NOT to have fixed set of arguments for `new_region` and then passing them around.

You can either manipulate ex_reg.dict directly or in some wrapper function per region, or add per-data-group methods to WebExport (and then call `ex_reg=ex.new_region(reg); ex_reg.add_johns_hopkins(hopkins_df.loc[code]); ...`)

Or consider column-group adding as suggested in https://github.com/epidemics/epimodel/pull/24#issuecomment-611973958 (most of those do not need any special formatting - they are either just columns of numbers, or date-sequence of numbers)

Reply:
> Or consider column-group adding as suggested in #24 (comment) (most of those do not need any special formatting - they are either just columns of numbers, or date-sequence of numbers)

I consider that, but from what I can see in the code now, it can be reused for [hopkins and foretold](https://github.com/epidemics/epimodel/pull/25/files#diff-45d92a1ec7e6ad88e92722302782b9dbR103-R112) at most. Not sure it's worth now to build an abstraction to reduce from 4 explicit custom wrangling to 3 (for the price of indirection in the code). If you think it's worth it, I can do it for these two.

> You can either manipulate ex_reg.dict directly or in some wrapper function per region, or add per-data-group methods to WebExport (and then call ex_reg=ex.new_region(reg); ex_reg.add_johns_hopkins(hopkins_df.loc[code]); ...)

That was my initial design and I scrapped that. WebExport doesn't need to touch any of this, so attaching it to this object is quite useless. We could use `*args, **kwargs` for that, but the various datasets needs custom handling, so I would have to pass something like `Tuple[pd.DataFrame, Callable]` where `Callable` would be the wrangling function generating the final dict, but that sounds overengineered. Lastly, I could put the dict creation of the web export, but then I am doing most of the export elsewhere (which I can, would be in a different module).

Reply:
:man_shrugging: Well, no use arguing, whatever works, I guess.

I discourage passing around long tuples of arguments, in a place where we may want to add or remove some.
I like abstractions for readability - if it is idiomatic, then you can read it as a sentece (e.g. `put_these_two_columns_there_for_all_relevant_regions()`).

Reply:
> I discourage passing around long tuples of arguments, in a place where we may want to add or remove some.

Then I need help then. [Here](https://github.com/epidemics/epimodel/pull/25/files#diff-d4579b2688d675235f402f6b4b43bcbfR59-R101) it's basically hardcoded - if you remove one of the datasets, it breaks. On the other, this function is providing very valuable insights thanks to which I was able to discover issues with indexing and joining. It also fails in situations which would lead to missing modeled data. Do you see how to make it so that it doesn't depend on adding datasets given there is some business logic around that? 

_If_ we agreed to have this one there, I don't see any drawbacks on having it in `new_region` too given that the code wouldn't run without that anyway. Or am I missing something?

Could I propose to build the abstraction when it actually becomes a problem, e.g. by adding additional 5 datasets or do you think it's so bad like this that it needs refactoring?